Remove Tutorial Text when resetting the level

Vhostym opened the pull request.

Vhostym committed:
Remove Tutorial Text when resetting the level

ambid17 commented on Assets/Scripts/Level/TutorialTriggerGroup.cs:
```diff
@@ -52,6 +52,7 @@ void UpdateTriggers()
             trigger.AddGameUI(gameUI);
             trigger.AddPlayerProgress(playerProgress);
         }
+        Instance = this;
```
why do you need this? It's set up in Awake()

Vhostym replied:
UpdateTriggers is called multiple times, and the awake version has gameUI, playerProgress and TutorialTriggers[] set to null and 0 respectively. In order for the instance to be usable it needs to be kept current. So this update is required.

ambid17 commented on Assets/Scripts/Level/TutorialTriggerGroup.cs:
```diff
@@ -61,5 +62,7 @@ public void ResetTriggers()
             trigger.ResetTrigger();
         }
+        string[] blankString = { };
+        gameUI.SetupTutorialTexts(blankString);
```
could just be `gameUI.SetupTutorialTexts(new string[] {});`

Vhostym replied:
Good point, was messing with declarations a bit to see whether I needed "" or {} and forgot to update.

Vhostym committed:
Reset now goes to Level Scriptable Object

Now when you reset the level the tutorial text provided is that done by the Level scriptable object, not nothing.

ambid17 committed:
move tutorialtriggergroup to player prefab

Vhostym committed:
Remove GetUIForColors due to obsolescence
